src/error.rs

```diff
@@ -67,16 +68,18 @@ impl From<ec_gpu_gen::EcError> for Error {
     }
 }
 
-impl error::Error for Error {}
+#[cfg(feature = "std")]
+impl std::error::Error for Error {}
 
-impl fmt::Display for Error {
+impl core::fmt::Display for Error {
```

Comment:
```suggestion
impl fmt::Display for Error {
```

This will work thanks to your top-level import of `core::fmt`.